design/20241124-rename-bunde-to-clusterbundle.md

```diff
@@ -0,0 +1,252 @@
+# Design: Renaming Bundle to ClusterBundle
+
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+- [Alternatives](#alternatives)
+- [Future Work](#future-work)
+
+## Release Signoff Checklist
+
+This checklist contains actions which must be completed before a PR implementing this design can be
+merged.
+
+- [ ] This design doc has been discussed and approved
+- [ ] Test plan has been agreed upon and the tests implemented
+- [ ] User-facing documentation has been PR-ed against the release branch in [cert-manager/website]
+
+TODO: add more items to checklist
+
+## Summary
+
+We want to rename `Bundle` to `ClusterBundle` for reasons described in "Motivation" below.
+
+Conversion webhooks in Kubernetes can only be used to convert between versions of an API,
+and not to convert between resources. This makes renaming harder to do, as it will represent
+deleting the old API and creating a new API.
+
+Even though trust-manager `Bundle` API is officially in an alpha state, we want to provide a smooth
+migration path for our users. For this reason we are proposing to do this change over multiple releases,
+requiring end-user action on the way, but without risking trust-manager target resources - which can provide
+critical features in a cluster. For additional details, please see "Proposal" below.
+
+## Motivation
+
+We have [an open issue](https://github.com/cert-manager/trust-manager/issues/63) suggesting renaming Bundle to ClusterBundle.
+Might seem like an obvious WontFix, considering all the assumed work (and pain) to do this,
+but it turns out that the current name somehow blocks introducing an equivalent namespace-scoped resource -
+for trust bundle management inside namespaces in multi-tenant clusters.
+
+In addition, `ClusterBundle` would be a better name looking at cert-manager having `Issuer` and `ClusterIssuer`.
+The name would also be better aligned with upstream [ClusterTrustBundle](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#cluster-trust-bundles).
+
+By pushing users to migrate smoothly to a new API, we can perform requested API changes at the same time.
+Consider it similar to a one-way conversion webhook.
+
+Since trust-manager is maturing and getting more attention, we suggest establishing its own API group: `trust-manager.io`.
+
+### Goals
+
+- `Bundle` resource is renamed to `ClusterBundle`
+- `ClusterBundle` is created in the new `trust-manager.io` API group
```

Comment:
to trust-manager.io from cert-manager.io?

Reply:
Yes, from `trust.cert-manager.io` to `trust-manager.io` - as a grown-up. 😉